app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorEndToEnd.kt

```diff
@@ -0,0 +1,251 @@
+    composeTestRule.onNodeWithContentDescription("Profile Icon").assertExists()
+    composeTestRule.onNodeWithTag("Find a Student").performClick()
+    composeTestRule.onNodeWithTag("screenTitle").assertExists()
+    composeTestRule.onNodeWithText("Physics Tutoring").assertIsDisplayed()
+    composeTestRule.onNodeWithText("physics").assertIsDisplayed()
+    composeTestRule.onNodeWithText("physics").performClick()
+    Thread.sleep(5000)
+    composeTestRule.onNodeWithTag("tutorLessonResponseScreen").assertExists()
+    composeTestRule.onNodeWithText("Ozymandias Halifax").assertExists()
+    composeTestRule.onNodeWithText("Offer to Teach (50.-/hour)").assertExists()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+    composeTestRule
+        .onNodeWithText(
+            "Would you like to offer to teach this lesson at your standard rate of 50.-/hour?")
+        .assertExists()
+    composeTestRule.onNodeWithTag("confirmLessonButton").performClick()
+    composeTestRule.onNodeWithContentDescription("Profile Icon").assertExists()
+    composeTestRule
+        .onNodeWithTag("section_Waiting for the Student Confirmation")
+        .assertIsDisplayed()
+    composeTestRule.onNodeWithTag("lessonCard_0").assertExists()
+
+    mockLessons =
+        listOf(
+            Lesson(
+                id = "1",
+                title = "Maths Tutoring",
+                description = "Fourrier Transform",
+                subject = Subject.ANALYSIS,
+                languages = listOf(Language.ENGLISH),
+                tutorUid = listOf("mockUid"),
+                studentUid = "student123",
+                minPrice = 20.0,
+                maxPrice = 50.0,
+                timeSlot = "16/11/2024T12:00:00",
+                status = LessonStatus.PENDING_TUTOR_CONFIRMATION,
+                latitude = 46.518973490411526,
+                longitude = 6.5685102716088295),
+        )
+    // Call the updatelesson
+    val updatedMockLessonFlow = MutableStateFlow(mockLessons[0])
+    doReturn(updatedMockLessonFlow).`when`(mockLessonViewModel).selectedLesson
+    // Reload the screen
+    composeTestRule.onNodeWithContentDescription("Profile Icon").performClick()
+    composeTestRule.onNodeWithTag("closeButton").performClick()
+    //
+    composeTestRule.onNodeWithTag("section_Waiting for your Confirmation").assertExists()
+    composeTestRule.onNodeWithTag("lessonCard_0").assertExists()
+    composeTestRule.onNodeWithText("Maths Tutoring").assertIsDisplayed()
+    composeTestRule.onNodeWithText("analysis").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("lessonCard_0").performClick()
+    composeTestRule.onNodeWithText("Ozymandias Halifax").assertExists()
+
+    composeTestRule.onNodeWithTag("cancelButton").performClick()
+    composeTestRule.onNodeWithText("Are you sure you want to dismiss this lesson?").assertExists()
+    composeTestRule.onNodeWithText("Cancel").performClick()
```

Comment:
Would probably be good to also test actually deleting the lesson, to see if that functions properly.